core/src/main/java/tc/oc/pgm/flag/post/SinglePost.java

```diff
@@ -0,0 +1,182 @@
+package tc.oc.pgm.flag.post;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.collect.ImmutableList;
+import java.time.Duration;
+import java.util.Random;
+import javax.annotation.Nullable;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import tc.oc.pgm.api.feature.FeatureReference;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.flag.Flag;
+import tc.oc.pgm.points.AngleProvider;
+import tc.oc.pgm.points.PointProvider;
+import tc.oc.pgm.points.PointProviderLocation;
+import tc.oc.pgm.teams.TeamFactory;
+
+public class SinglePost extends PostDefinition {
+
+  private static final int MAX_SPAWN_ATTEMPTS = 100;
+
+  private final @Nullable FeatureReference<TeamFactory>
+      owner; // Team that owns the post, affects various things
+  private final Duration
+      recoverTime; // Time between a flag dropping and being recovered, can be infinite
+  private final @Nullable Duration
+      respawnTime; // Fixed time between a flag being recovered and respawning at the post
+  private final @Nullable Double
+      respawnSpeed; // Makes respawn time proportional to distance, flag "moves" back at this m/s
+  private final ImmutableList<PointProvider> returnPoints; // Spawn points for the flag
+  private final boolean
+      sequential; // Search for spawn points sequentially, see equivalent field in SpawnInfo
+  private final boolean permanent; // Flag enters Completed state when at this post
+  private final double pointsPerSecond; // Points awarded while any flag is at this post
+  private final Filter pickupFilter; // Filter players who can pickup a flag at this post
+  private final Filter respawnFilter; // Filter if a flag can respawn to this post
+  private final @Nullable String
+      postName; // The post's name, shown in chat when the flag is respawning
```

Comment:
Only visual: Maybe just have the comments above the variables instead of having an indent between the variable type and the variable name.

```java
//Comment
private final Variable name
```
instead of 
```java
private final Variable
    name //Comment
```

Reply:
It'd look better if it was just not having a line-jump, but sadly the formatter enforces that sooo...
i may move them all to above the var